sensors: lsm6dsv16x: fix double-promotion warning

kartben opened the pull request:
use roundf instead of round in Q31_SHIFT_VAL macro to avoid
    double-promotion warning


```
/__w/zephyr/zephyr/drivers/sensor/st/lsm6dsv16x/lsm6dsv16x_decoder.c:459:29: error: implicit conversion increases floating-point precision: 'float32_t' (aka 'float') to 'double' [-Werror,-Wdouble-promotion]
                        out->readings[count].x = Q31_SHIFT_VAL(x.f, out->shift);
                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

teburd reviewed (changes requested):
roundf() instead to fix this?

Most devices don't have any hardware to help with doubles

For example how bad this gets on esp32...
https://www.reddit.com/r/esp32/comments/t46960/comment/hyz06fb/

kartben commented:
> roundf() instead to fix this?
> 
> Most devices don't have any hardware to help with doubles
> 
> For example how bad this gets on esp32... [reddit.com/r/esp32/comments/t46960/comment/hyz06fb](https://www.reddit.com/r/esp32/comments/t46960/comment/hyz06fb/)

Mmm ok -- this was going to be my fix initially but wasn't sure this was any better based on the (non-)usage in-tree

```
git grep roundf | wc -l
3
```

teburd commented:
> > roundf() instead to fix this?
> > Most devices don't have any hardware to help with doubles
> > For example how bad this gets on esp32... [reddit.com/r/esp32/comments/t46960/comment/hyz06fb](https://www.reddit.com/r/esp32/comments/t46960/comment/hyz06fb/)
> 
> Mmm ok -- this was going to be my fix initially but wasn't sure this was any better based on the (non-)usage in-tree
> 
> ```
> git grep roundf | wc -l
> 3
> ```

Well that's a shocking issue for a different reason because as noted here... double mults are going to be very slow in comparison.

There's not tons of round()'s either but those should imho be reviewed...

Suspicious usages of double math...

drivers/video/ov5640.c
drivers/dai/intel/dmic/dmic.h
soc/nordic/nrf54l/soc.c

kartben committed:
sensors: lsm6dsv16x: fix double-promotion warning

use roundf instead of round in Q31_SHIFT_VAL macro to avoid
double-promotion warning

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

avisconti reviewed (commented):
saw a little bit late, sorry
